ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/channelbuilders/ReadOnlySeekableByteChannel.java

```diff
@@ -40,7 +40,7 @@ public ReadableByteChannel getChannel() {
     public int read(final ByteBuffer dst) throws IOException {
         size = channel.read(dst);
         position += size;
-        return size;
+        return Long.valueOf(size).intValue();
```

Comment:
Did this cause problems when a blob was over 4GB?

Reply:
This is one of the places it failed, I can't say if this is wholly the issue though.